File: Benchmarking/src/gen_108_15.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: [s0]
  }
  @inline(never)
  func f0(_ v0: [s1], _ v1: [[s1]], _ v2: Double) -> Double {
    let v7: s1 = v0[2]
    let v3: s0 = v7.p0
    let v20: [[Double]] = v3.p1
    var v10: s0 = v3
    let v15: [[Double]] = v3.p1
    v10.p1 = v15
    let v18: [[Double]] = v10.p0
    v10.p1 = v15
    let v23: [Double] = v18[0]
    v10.p1 = v20
    let v47: Double = v23[0]
    return v47
  }
  func benchmark() {
    let v0: [s1] = [s1(p0: s0(p0: [[0.0], [1.0], [2.0]], p1: [[3.0], [4.0]]), p1: [s0(p0: [[5.0], [6.0], [7.0]], p1: [[8.0], [9.0]])]), s1(p0: s0(p0: [[10.0], [11.0], [12.0]], p1: [[13.0], [14.0]]), p1: [s0(p0: [[15.0], [16.0], [17.0]], p1: [[18.0], [19.0]])]), s1(p0: s0(p0: [[20.0], [21.0], [22.0]], p1: [[23.0], [24.0]]), p1: [s0(p0: [[25.0], [26.0], [27.0]], p1: [[28.0], [29.0]])])]
    let v1: [[s1]] = [[s1(p0: s0(p0: [[30.0], [31.0], [32.0]], p1: [[33.0], [34.0]]), p1: [s0(p0: [[35.0], [36.0], [37.0]], p1: [[38.0], [39.0]])])]]
    let v2: Double = 40.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 40.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_108_15.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: s0,
    p1: Vector[s0]
  )
  @noinline
  def f0(v0: Vector[s1], v1: Vector[Vector[s1]], v2: Double): Double = {
    val v7: s1 = v0(2)
    val v3: s0 = v7.p0
    val v20: Vector[Vector[Double]] = v3.p1
    var v10: s0 = v3
    val v15: Vector[Vector[Double]] = v3.p1
    v10 = v10.copy(p1 = v15)
    val v18: Vector[Vector[Double]] = v10.p0
    v10 = v10.copy(p1 = v15)
    val v23: Vector[Double] = v18(0)
    v10 = v10.copy(p1 = v20)
    val v47: Double = v23(0)
    v47
  }
  def benchmark(): Unit = {
    val v0: Vector[s1] = Vector(s1(s0(Vector(Vector(0.0), Vector(1.0), Vector(2.0)), Vector(Vector(3.0), Vector(4.0))), Vector(s0(Vector(Vector(5.0), Vector(6.0), Vector(7.0)), Vector(Vector(8.0), Vector(9.0))))), s1(s0(Vector(Vector(10.0), Vector(11.0), Vector(12.0)), Vector(Vector(13.0), Vector(14.0))), Vector(s0(Vector(Vector(15.0), Vector(16.0), Vector(17.0)), Vector(Vector(18.0), Vector(19.0))))), s1(s0(Vector(Vector(20.0), Vector(21.0), Vector(22.0)), Vector(Vector(23.0), Vector(24.0))), Vector(s0(Vector(Vector(25.0), Vector(26.0), Vector(27.0)), Vector(Vector(28.0), Vector(29.0))))))
    val v1: Vector[Vector[s1]] = Vector(Vector(s1(s0(Vector(Vector(30.0), Vector(31.0), Vector(32.0)), Vector(Vector(33.0), Vector(34.0))), Vector(s0(Vector(Vector(35.0), Vector(36.0), Vector(37.0)), Vector(Vector(38.0), Vector(39.0)))))))
    val v2: Double = 40.0
    val start = nanoTime()
    var result: Double = 40.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1, v2)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}